sourcetracker/tests/test_sourcetracker.py

```diff
@@ -10,119 +10,156 @@
-        samples = ['s4', 's2', 's0', 's1']
-        sort = True
-        obs_envs, obs_collapsed_sources = collapse_sources(samples,
-                                                           sample_metadata,
-                                                           category,
-                                                           biom_table, sort)
-        exp_envs = np.array(['1', '2'])
-        exp_collapsed_sources = \
-            np.array([[4, 14, 24, 34, 44, 54, 64, 74, 84, 94],
-                      [3, 13, 23, 33, 43, 53, 63, 73, 83, 93]])
-        np.testing.assert_array_equal(obs_collapsed_sources,
-                                      exp_collapsed_sources)
-        np.testing.assert_array_equal(obs_envs, exp_envs)
-
-        category = 'cat2'
-        samples = ['s4', 's2', 's0', 's1']
-        sort = True
-        obs_envs, obs_collapsed_sources = collapse_sources(samples,
-                                                           sample_metadata,
-                                                           category,
-                                                           biom_table, sort)
-        exp_envs = np.array(['B', 'C', 'D'])
-        exp_collapsed_sources = \
-            np.array([[0, 5, 10, 15, 20, 25, 30, 35, 40, 45],
-                      [1, 6, 11, 16, 21, 26, 31, 36, 41, 46],
-                      [6, 16, 26, 36, 46, 56, 66, 76, 86, 96]])
-        np.testing.assert_array_equal(obs_collapsed_sources,
-                                      exp_collapsed_sources)
-        np.testing.assert_array_equal(obs_envs, exp_envs)
+    def test_check_and_correct_data(self):
+        # A table where nothing is wrong, no changes expected.
+        data = np.random.randint(0, 100, size=18).reshape(3, 6)
+        ftable = pd.DataFrame(data)
+        obs = check_and_correct_data(ftable, 'True')
+        pd.util.testing.assert_frame_equal(ftable, obs)
+
+        _ftable = ftable.copy()
+
+        # A table with nans, expect a ValueError.
+        _ftable.iloc[0, 3] = np.nan
+        self.assertRaises(ValueError, check_and_correct_data, _ftable, True)
+
+        # Check that fractional counts give an error if function isn't allowed
+        # to correct them.
+        _ftable.iloc[0, 3] = 4.512
+        self.assertRaises(ValueError, check_and_correct_data, _ftable, False)
+
+        # Check that fractional counts are corrected.
+        ftable.iloc[0, 3] = 5
+        _ftable.iloc[0, 3] = 4.512
+        obs = check_and_correct_data(_ftable, True)
+        pd.util.testing.assert_frame_equal(ftable, obs)
+
+    def test_biom_to_df(self):
+        data = np.arange(200).reshape(20, 10).astype(float)
+        data2 = data.copy()
+        oids = ['o%s' % i for i in range(20)]
+        sids = ['s%s' % i for i in range(10)]
 
+        obs = biom_to_df(Table(data, oids, sids))
+        exp = pd.DataFrame(data.astype(np.int64).T, index=sids, columns=oids)
+        pd.util.testing.assert_frame_equal(obs, exp)
```

Comment:
In general it's best to split different tests into different functions. Not a requirement for merge, but it's good practice. For example, if ``biom_to_df`` were modifying ``data`` in place, any subsequent tests that use that may or may not be doing what you expect.